cloudinit/sources/helpers/vultr.py

```diff
@@ -0,0 +1,293 @@
+            add = {
+                "type": "static6",
+                "control": "auto",
+                "address": additional['address'],
+                "netmask": additional['netmask']
+            }
+            netcfg['subnets'].append(add)
+
+    # Add config to template
+    return netcfg
+
+
+# Input Metadata and generate private network config part
+def generate_private_network_interface(md):
+    interface_name = get_interface_name(md['interfaces'][1]['mac'])
+    if not interface_name:
+        raise RuntimeError(
+            "Interface: %s could not be found on the system" %
+            md['interfaces'][1]['mac'])
+
+    netcfg = {
+        "name": interface_name,
+        "type": "physical",
+        "mac_address": md['interfaces'][1]['mac'],
+        "accept-ra": 1,
+        "subnets": [
+            {
+                "type": "static",
+                "control": "auto",
+                "address": md['interfaces'][1]['ipv4']['address'],
+                "netmask": md['interfaces'][1]['ipv4']['netmask']
+            }
+        ]
+    }
+
+    return netcfg
+
+
+# Generate the vendor config
+# This configuration is to replicate how
+# images are deployed on Vultr before Cloud-Init
+def generate_config(config):
+    LOG.debug("DS: %s", json.dumps(config))
+    md = get_cached_metadata(config)
+
+    # Grab the startup script
+    script = md['startup-script']
+
+    # Create vendor config
+    config_template = copy.deepcopy(md['vendor-config'])
+
+    # Add generated network parts
+    config_template['network'] = generate_network_config(config)
+
+    # Linux specific packages
+    if util.is_Linux():
+        config_template['packages'].append("ethtool")
+
+    # Define vendor script
+    vendor_script = "#!/bin/bash"
+
+    # Go through the interfaces
+    for netcfg in config_template['network']['config']:
+        # If the interface has a mac and is physical
+        if "mac_address" in netcfg and netcfg['type'] == "physical":
+            # Enable multi-queue on linux
+            # This is executed as a vendor script
+            if util.is_Linux():
+                # Set its multi-queue to num of cores as per RHEL Docs
+                name = netcfg['name']
+                command = "ethtool -L %s combined $(nproc --all)" % name
+                vendor_script = '%s\n%s' % (vendor_script, command)
+
+    # Add write_files if it is not present in the template
+    if 'write_files' not in config_template.keys():
+        config_template['write_files'] = []
+
+    # Add vendor script to config
+    config_template['write_files'].append(
+        {
+            'encoding': 'b64',
+            'content': convert_to_base64(vendor_script),
+            'owner': 'root:root',
+            'path': '/var/lib/scripts/vendor/vultr-interface-setup.sh',
+            'permissions': '0750'
+        }
+    )
+
+    # Write the startup script
+    if script and script != "echo No configured startup script":
```

Comment:
this is odd.  Does the metadata service actually pass that in as the null value ?

Reply:
Yeah, thats what is currently passed if there is no startup script.